src/main/java/seedu/address/model/eatery/Tag.java

```diff
@@ -0,0 +1,88 @@
+package seedu.address.model.eatery;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.HashMap;
+
+/**
+ * Represents an Eatery's tag in the EatMe application.
+ */
+public class Tag {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Tags should only contain alphabets and it should not be blank";
+    /*
+     * The first character of the tag must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "(^[A-Za-z]+[A-Za-z_]+[A-Za-z]$)";
+
+    private static int tagId = -1;
+    private static HashMap<String, Tag> tags = new HashMap<String, Tag>();
+
+    private int id;
+    private String name;
+
+    /**
+     * Constructs a {@code Tag}.
+     *
+     * @param name A valid tag name.
+     */
+    public Tag(String name) {
+        requireNonNull(name);
+        checkArgument(isValidTag(name), MESSAGE_CONSTRAINTS);
+        tagId = tagId + 1;
+        this.name = name;
+        this.id = tagId;
+    }
+
+    /**
+     * Returns a new Tag object if Tag with the same name doesn't exist,
+     * else it returns the existing Tag object.
+     */
+    public static Tag create(String name) {
+        Tag t = tags.get(name);
+
+        if (t == null) {
+            Tag newTag = new Tag(name);
+            tags.put(newTag.getName(), newTag);
+            return newTag;
+
+        } else {
+            return t;
+        }
```

Comment:
this is to check for duplicate tags? If im not wrong, this should be checked in the addCommand?

Reply:
????? i don't think it does hahahaha